src/coreclr/jit/llvm.cpp

```diff
@@ -2655,7 +2753,38 @@ void Llvm::lowerToShadowStack()
         _currentRange = &LIR::AsRange(_currentBlock);
         for (GenTree* node : CurrentRange())
         {
-            if (node->OperIs(GT_STORE_LCL_VAR, GT_LCL_VAR, GT_LCL_FLD, GT_LCL_VAR_ADDR, GT_LCL_FLD_ADDR))
+            if (node->OperIs(GT_STORE_LCL_VAR, GT_LCL_VAR, GT_LCL_VAR_ADDR))
+            {
+                // fail unsupported promoted structs
+                GenTreeLclVarCommon* lclVar = node->AsLclVarCommon();
+                LclVarDsc*           varDsc = _compiler->lvaGetDesc(lclVar->GetLclNum());
+
+                if (_compiler->lvaIsFieldOfDependentlyPromotedStruct(varDsc))
+                {
+                    switch(node->OperGet())
+                    {
+                        case GT_LCL_VAR:
+                            lclVar->ChangeOper(GT_LCL_FLD);
+                            break;
+
+                        case GT_STORE_LCL_VAR:
+                            lclVar->ChangeOper(GT_STORE_LCL_FLD);
+                            break;
+#
+                        case GT_LCL_VAR_ADDR:
+                            lclVar->ChangeOper(GT_LCL_FLD_ADDR);
+                            break;
+                    }
+                    lclVar->AsLclFld()->SetLclOffs(varDsc->lvFldOffset);
```

Comment:
The local number needs to be set to the parent as well.

Additionally, there is some flag updating that needs to be done. `LCL_[FLD|VAR]_ADDR` and `STORE_LCL_[VAR|FLD]` need to gain `GTF_VAR_USEASG` if they're `IsPartialLclFld` and had `GTF_VAR_DEF` set.

Reply:
Thanks, in https://github.com/dotnet/runtimelab/pull/2007/commits/df05ac468ac2df80292e0cd739f1d2dfe09dd27f